platform/mellanox/mlnx-platform-api/sonic_platform/fan.py

```diff
@@ -15,23 +15,28 @@
 except ImportError as e:
     raise ImportError (str(e) + "- required module not found")
 
-LED_ON = 1
-LED_OFF = 0
+LED_ON = '1'
+LED_OFF = '0'
 
 PWM_MAX = 255
 
 FAN_PATH = "/var/run/hw-management/thermal/"
 LED_PATH = "/var/run/hw-management/led/"
+# fan_dir only exist Spectrum 2 switches
```

Comment:
Please change the comment that it is currently not supported in spectrum 1 and later new Silicon should not be addressed and will have it supported by default

Reply:
fixed.